services/eventhistory/pkg/service/service.go

```diff
@@ -75,3 +76,39 @@ func (eh *EventHistoryService) GetEvents(ctx context.Context, req *ehsvc.GetEven
 
 	return nil
 }
+
+// GetEventsForUser allows to retrieve events from the eventstore by userID
+// This function will match all events that contains the user ID between two non-word characters.
+// The reasoning behind this is that events put the userID in many different fields, which can differ
+// per event type. This function will match all events that contain the userID by using a regex.
+// This should also cover future events that might contain the userID in a different field.
+func (eh *EventHistoryService) GetEventsForUser(ctx context.Context, req *ehsvc.GetEventsForUserRequest, resp *ehsvc.GetEventsResponse) error {
+	idx, err := eh.store.List(store.ListPrefix(""))
+	if err != nil {
+		eh.log.Error().Err(err).Msg("could not list events")
+		return err
+	}
+
+	// Match all events that contains the user ID between two non-word characters.
+	userID := regexp.MustCompile(fmt.Sprintf(`\W%s\W`, req.UserID))
```

Comment:
Better not use `Must` here. `req.UserId` is coming from outside, it could contain characters that break the regexp, which would then lead to a panic.